app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,21 @@
+class SessionsController < ApplicationController
+  def new; end
+
+  def create
+    user = User.find_by email: params.dig(:session, :email)&.downcase
+    if user&.authenticate params.dig(:session, :password)
```

Comment:
Toán tử safe navigator `&.` có tác dụng gì?

Reply:
Dạ toán tử `&.` giúp truy cập thuộc tính hoặc phương thức của một đối tượng một cách an toàn, kiểm tra xem đối tượng có tồn tại hay không trước khi thực hiện thao tác. Nếu đối tượng không tồn tại (là nil), nó sẽ trả về nil thay vì ném ra một lỗi.